runtime/src/main/java/com/datastax/oss/quarkus/runtime/graal/NativeSubstitutions.java

```diff
@@ -15,20 +15,40 @@
  */
 package com.datastax.oss.quarkus.runtime.graal;
 
+import com.datastax.oss.driver.internal.core.os.CpuInfo;
 import com.datastax.oss.driver.internal.core.os.Native;
+import com.oracle.svm.core.annotate.KeepOriginal;
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
 
 @TargetClass(Native.class)
+@Substitute
 final class NativeSubstitutions {
 
-  /**
-   * This method returns always false because jnr is not available.
-   *
-   * @return false denoting that {@link Native#getProcessId()} cannot be called.
-   */
+  @Substitute private static final CpuInfo.Cpu CPU = CpuInfo.determineCpu();
+
   @Substitute
   public static boolean isGetProcessIdAvailable() {
     return false;
   }
+
+  @Substitute
+  public static int getProcessId() {
+    throw new RuntimeException("Native operations are not supported on this platform");
+  }
+
+  @Substitute
+  public static boolean isCurrentTimeMicrosAvailable() {
+    return false;
+  }
+
+  @Substitute
+  public static long currentTimeMicros() {
+    throw new RuntimeException("Native operations are not supported on this platform");
+  }
```

Comment:
If you are disabling native clock this way, I think you can remove this:
`configLoaderBuilder.withBoolean(DefaultDriverOption.TIMESTAMP_GENERATOR_FORCE_JAVA_CLOCK, true);`
from
`AbstractCassandraClientProducer`

Reply:
I agree, but I'd argue this should be done as part of the changes when we get a driver that contains JAVA-2663.  When that happens we can remove this restriction _and_ remove the NativeSubstitutions entirely.

Reply:
Just occurred to me: it probably makes sense to have a discrete ticket to represent the work referenced above.

Reply:
JAVA-2789 has been created to address this work